frappe/core/doctype/report/test_report.py

```diff
@@ -145,7 +145,7 @@ def test_report_with_custom_column(self):
 					"fieldtype": "Data",
 					"label": "Email",
 					"insert_after_index": 0,
-					"link_field": "name",
+					"link_field": {"fieldname": "name", "names": []},
```

Comment:
Just confirming, will the previously saved report with string instead of a dict continue to work?

Reply:
@ankush 
Removed changes on `test_report.py`.

`add_custom_column_data` and `get_data_for_custom_report` have been refactored to be backward compatible. Performance Issue will be fixed on both old and newly created Custom Reports.